cltk/lemmatize/latin/unsupervised.py

```diff
@@ -0,0 +1,382 @@
+        -------
+        none'''
+        tessobj = TessFile(filepath)
+        tokengenerator = iter(tessobj.read_tokens())
+        stop = 0
+        while stop != 1:
+            try:
+                rawtoken = next(tokengenerator)
+                cleantoken_list = self.token_cleanup(rawtoken)
+                token = cleantoken_list[0]
+                self.countgram(token)
+            except StopIteration:
+                stop = 1
+
+    def countgram(self, targettoken):
+        '''Update the frequency model with a new token from the corpus.'''
+        lemmas = self.lemmatizer.lookup([targettoken])
+        lemmas = self.lemmatizer.isolate(lemmas)
+        for lem in lemmas:
+            try:
+                test_presence = self.count_dictionary[lem]
+            except KeyError:
+                self.count_dictionary[lem] = 0
+            self.count_dictionary[lem] += 1
+
+    def lemmatize(self, target):
+        '''Use the unsupervised count of lemma frequencies generated by read_files()
+        to assign probabilities in the case of an ambiguous lemmatization.
+        parameters
+        ----------
+        target: a token to be lemmatized
+        results
+        -------
+        a list of tuples of the form [(lemma, probability)]
+        '''
+        if target in self.punctuation_list:
+            lemmalist = [('punc', 1)]
+            return lemmalist
+        if target == 'ne':
+            lemmalist = [('ne', 1)]
+            return lemmalist
+        lemmalist = self.lemmatizer.lookup([target])
+        lemmas = self.lemmatizer.isolate(lemmalist)
+        if len(lemmas) > 1:
+            all_lemmas_total = sum([self.count_dictionary[l] for l in lemmas])
+            try:
+                lemmalist = [(l, (self.count_dictionary[l] / all_lemmas_total)) for l in lemmas]
+            except ZeroDivisionError:
+                print([(self.count_dictionary[l], l) for l in lemmas])
+            return lemmalist
+        lemmalist = []
+        lemmaobj = (lemmas[0], 1)
+        lemmalist.append(lemmaobj)
+        return lemmalist
+
+    def token_cleanup(self, rawtoken):
+        '''Standardize tokens by replaceing j with i and v with u, and
+        split into multiple tokens as needed with tokenize() method of word_tokenizer class
+        parameters
+        ----------
+        rawtoken: the token as drawn from the text
+        return
+        ------
+        tokenlist: a list of possible word or punctuation tokens
+        '''
+        rawtoken = self.jv.replace(rawtoken)
+        rawtoken = rawtoken.lower()
+        tokenlist = self.word_tokenizer.tokenize(rawtoken)
+        #sometimes words are split into enclitics and punctuation.
+        return tokenlist
+
+    def save_pickle(self, filename):
+        '''Saves the self.count_dictionary object for later reuse.
+        dependencies
+        ------------
+        os package
+        parameters
+        ----------
+        filename: name for the pickle file'''
```

Comment:
Please place multiple line triple quote on its own line, and have a space separating the comment from the code section, otherwise it's hard to read. Et ubicumque...